### PR TITLE
feat(lint): split actions in linting and regression

### PR DESCRIPTION
**Pull request by fzipi**
Executing linting on every commit makes more sense, and then regression only on tests or rules changed only

Signed-off-by: Felipe Zipitria <felipe.zipitria@owasp.org>

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
+name: Lint
+
+on: [push, pull_request]
+
+jobs:
+  check-syntax:
+    runs-on: ubuntu-latest
+    strategy:
+      # change to true
```
Comment (fgsch) [inline .github/workflows/lint.yaml:10]: Outdated comment?

File: .yamllint.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+rules:
+  # Test lines can be big
+  line-length:
+    max: 1024
+    level: warning
+    ignore: |
+      tests/regression/tests/REQUEST-920-PROTOCOL-ENFORCEMENT/920380.yaml
+      tests/regression/tests/REQUEST-920-PROTOCOL-ENFORCEMENT/920390.yaml
+      tests/regression/tests/REQUEST-941-APPLICATION-ATTACK-XSS/941360.yaml
```
Comment (fgsch) [inline .yamllint.yml:11]: Why are we ignoring these?

Comment (fgsch) [reply .yamllint.yml:11]: Can we add a comment to document the reason?

**Comment by fgsch** [comment]
I left a few comments but looks good.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -14,33 +14,6 @@ on:
...
-
-      - name: "Check syntax"
-        run: |
-          python -V
-          pip install --upgrade setuptools
-          pip install -r tests/integration/requirements.txt
-          git clone https://github.com/CRS-support/secrules_parsing
-          pip install -r secrules_parsing/requirements.txt
-          python secrules_parsing/secrules_parser.py -c -f rules/*.conf
-          py.test -vs tests/integration/format_tests.py
```
Comment (fgsch) [inline .github/workflows/test.yml:41]: IIUC we can kill this test now since we're using the other linter.

Comment (fgsch) [reply .github/workflows/test.yml:41]: This is a question, not something we need to do in this PR.

File: .linelint.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+rules:
+  # checks if file ends in a newline character
+  end-of-file:
+    # set to true to enable this rule
+    enable: true
+
+    # set to true to disable autofix (if enabled globally)
+    disable-autofix: false
```
Comment (fgsch) [inline .linelint.yml:8]: Is there any point to have this enabled here?

Comment (fzipi) [reply .linelint.yml:8]: This one replaces the one in `py.test -vs tests/integration/format_tests.py`. If we don't wan't that at all, we can remove everything.

File: .github/workflows/test.yml (hunk 3 above)
Comment (fzipi) [reply .github/workflows/test.yml:41]: We should, but we didn't change it yet.

File: .linelint.yml (hunk 4 above)
Comment (fgsch) [reply .linelint.yml:8]: I was referring to the `disable-autofix`. Not sure we're talking about the same.

Comment (fzipi) [reply .linelint.yml]: Yeah, now I got it. Fixing

File: .yamllint.yml (hunk 2 above)
Comment (fzipi) [reply .yamllint.yml:13]: Done.

File: .github/workflows/lint.yaml (hunk 1 above)
Comment (fzipi) [reply .github/workflows/lint.yaml]: Copy & paste.

**Comment by fgsch** [comment]
LGTM!

**Pull request merged**